### ISSUE TITLE
How to optimize my dataloader 

### ISSUE THREAD

**Issue by NomiMalik0207** [pytorch/pytorch#15536]
I have dataloader code below. my data is of 200gb and for one batch it takes 5 sec to laod. this made gpu waiting for thimport torch
import numpy as np
import os, random
from PIL import ImageFile

ImageFile.LOAD_TRUNCATED_IMAGES = True
from torch.utils.data import Dataset
from torch import nn
import torchvision.transforms as transforms
import torchvision.utils as vutils
import time
import easydict
import matplotlib.pyplot as plt
from PIL import Image
import os
import imageio
from torch.autograd import Variable

channel_num = 3
img_size = 96
Nd = 0
Np = 0
Nz = 50
multiPIE_train_transform = transforms.Compose([
    transforms.CenterCrop(160),
    transforms.Resize(img_size),
    transforms.ToTensor(),
    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
])

def PIL_list_reader(fileList):
    imgList = [];
    idList = [];
    poseList = [];
    with open(fileList, 'r') as file:
        for line in file.readlines():
            imgPath, id_label, pose_label = line.strip().rstrip('\n').split(' ')
            # use
            imgList.append(imgPath)
            idList.append(int(id_label.encode("utf-8")))
            poseList.append(int(pose_label.encode("utf-8")))

    Np = int(max(poseList) + 1)
    Nd = int(max(idList) + 1)

    return [imgList, poseList, idList, Np, Nd]

class multiPIE(Dataset):
    def __init__(self, fileList,transform_mode=None, list_reader=PIL_list_reader):
        self.channel_num = channel_num
        self.image_size = img_size
        self.Nz = 50
        self.imgList, self.pose_label, self.id_label, self.Np, self.Nd = list_reader(fileList)

        if transform_mode == 'train_transform':
           self.transform = multiPIE_train_transform

    def __getitem__(self, index):
    # start=time.time()
        imgPath = self.imgList[index]
        start = time.time()
        img = Image.open(imgPath)
        print("time for loading image", time.time() - start)
        if self.transform is not None:
            img = self.transform(img)
        
        return img, self.id_label[index], self.pose_label[index]
    def __len__(self):
       return len(self.imgList)

args = easydict.EasyDict({
  
    "batch_size": 64,
    "save_freq": 1,
    "data_path": "/home/naveed/myDRGAN-master/data/imagessingle.txt",
    " Np": 9,
    "Nd": 200,
    "Nd": 50,
})
device = torch.device("cuda:0")

dataset = multiPIE(args.data_path, transform_mode='train_transform')
dataloader = torch.utils.data.DataLoader(dataset, batch_size=args.batch_size, shuffle=True, num_workers=0)
start = time.time()
realBatch = next(iter(dataloader))
a = realBatch[0]
print("time for data loading", time.time() - start)
e next batch. How can i optimize my dataloader.

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions
